src/ServiceMonitor/IISConfigUtil.cpp

```diff
@@ -121,10 +121,9 @@ HRESULT IISConfigUtil::Initialize()
     return hr;
 }
 
-HRESULT IISConfigUtil::BuildAppCmdCommand(unordered_map<wstring, wstring> envSet, WCHAR* pstrAppPoolName, wstring** pStrCmd, BOOL fAddCommand)
+HRESULT IISConfigUtil::BuildAppCmdCommand(vector<pair<wstring, wstring>> vecSet, WCHAR* pstrAppPoolName, wstring** pStrCmd, BOOL fAddCommand, int* beginIndex)

```

Comment:
Did you think about using a typedef to make this contract a little more clear when calling?

```
typedef long APPCMD_CMD_TYPE

APPCMD_CMD_TYPE APPCMD_ADD = 0;
APPCMD_CMD_TYPE APPCMD_REMOVE = 1;
```